[core]/es_extended/server/main.lua

```diff
@@ -628,6 +628,22 @@ ESX.RegisterServerCallback('esx:getPlayerNames', function(source, cb, players)
   cb(players)
 end)
 
+ESX.RegisterServerCallback("esx:spawnVehicle", function(source,cb,model,coords,heading,props,warp)
+  local ped = GetPlayerPed(source)
+  local vehicle = nil
+  props = props or {}
+  ESX.OneSync.SpawnVehicle(model, coords, heading, props, function(id)
+    vehicle = NetworkGetEntityFromNetworkId(id)
+    if warp then
+      while GetVehiclePedIsIn(ped) ~= vehicle do
```

Comment:
If will never be equal will run forever?

This should be an attempt for say 15 if it fails then break.